src/llm/response_parsers/qwen3_response_parser.hpp

```diff
@@ -64,9 +71,14 @@ class Qwen3ResponseParser : public BaseResponseParser {
         // Assuming content ends when tool calls start, so we find the first occurrence of <tool_call> after the content start
         auto contentEndIt = std::find(contentStartIt, generatedTokens.end(), toolCallStartTokenId);
 
-        if (contentStartIt != generatedTokens.end() && contentEndIt != generatedTokens.end() && contentStartIt < contentEndIt) {
+        if (contentStartIt == contentEndIt) {
```

Comment:
can we already hide implementation inside cpp unit? what was the explanation of why not earlier? this will greatly decrease compilation time

Reply:
if we did that, we probably could even hide rapidjson includes from headers. please dont extend already long compile times

Reply:
please do that for all new tool parsers